### PR TITLE
Systrace ETW Tracing, TraceLogging instead of manifest based events

### PR DESCRIPTION
**Pull request by mganandraj**
Last year, we had plumbed the Systrace sections and events in the core react native to windows code and fired ETW events. But, it was hard for people to use because we used manifest based ETW events. This change converts the event triggering code to use Tracelogging APIs (aka self describing events) which doesn't require the manifest to be registered on the host machine to inspect traces.

Post this change, the systrace events can be inspected on any windows machines where the RNW based apps are running by issuing the following commands,

wpr.exe -start rnw.wprp
wpr.exe -stop rnw.etl
wpa.exe rnw.etl

This works for Win32 apps too, say Office.

I'll look into tooling to make to real time view the traces, like 'timber' app in Office. Also, it will be useful to expose a logging API over this ETW infrastructure.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6869)

### CODE REVIEW

**Review by stecrain — APPROVED** [review]

File: vnext/Shared/tracing/cmd.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+REM Enable RNW provider/collector along side general system event providers.
+wpr.exe -start GeneralProfile -start rnw.wprp
+wpr.exe -start GeneralProfile -start rnw.etl
+
+REM Enable only RNW provider/collector.
+wpr.exe -start rnw.wprp
+wpr.exe -stop rnw.etl
```
Comment (asklar) [inline vnext/Shared/tracing/cmd.txt:7]: consider moving this to a PS script in vnext\scripts

Can you add a small how-to in the wiki?

**Review by asklar — APPROVED** [review]
thanks for doing this!

**Comment by adam-bratin** [comment]
From my understanding there is one drawback from moving from manifest based ETW events to LoggingChannels (or newer ETW mechanisms). A developer can no longer register with the OS their provider so that logging can be done during MS OOBE. Now I agree that for most users that is not an issue, but for some applications that run during that time period this is a vital tool for bug isolation and performance benchmarking. I just wanted to bring up that tradeoff. Especially as a developer that works on an application in that space, who has spent many hours combing through ETL files and running OOBE boot tests.

File: vnext/Shared/tracing/cmd.txt (hunk 1 above)
Comment (mganandraj) [reply vnext/Shared/tracing/cmd.txt:7]: Hi @asklar,

I've now added PS scripts in vnext\scripts\Tracing\ to manage the sessions.

Regarding the how-to documentation, where should I add the information ? Is is under top-level docs folder in the repo ? Should i update an existing file or add a new "tracing.md" ?

Thanks!

**Comment by mganandraj** [comment]
> From my understanding there is one drawback from moving from manifest based ETW events to LoggingChannels (or newer ETW mechanisms). A developer can no longer register with the OS their provider so that logging can be done during MS OOBE. Now I agree that for most users that is not an issue, but for some applications that run during that time period this is a vital tool for bug isolation and performance benchmarking. I just wanted to bring up that tradeoff. Especially as a developer that works on an application in that space, who has spent many hours combing through ETL files and running OOBE boot tests.

Hi @adam-bratin , Thanks for sharing this nuance. Will keep it in record !

File: vnext/Shared/tracing/cmd.txt (hunk 1 above)
Comment (asklar) [reply vnext/Shared/tracing/cmd.txt:7]: great thanks! we have some docs under \docs and we also have a wiki:
image

I can see the how-to becoming bigger as we continue to invest in perf measurement and tooling, and eventually make this available/easily consumable to 3rd parties, so we will probably want to _eventually_ make it part of the docs that live on the RNW website (https://microsoft.github.io/react-native-windows/) but for now it's fine to keep in the wiki (or docs if you prefer), and we can move it once it's meaty enough

Comment (mganandraj) [reply vnext/Shared/tracing/cmd.txt]: Added a rudimentary documentation to start with .. https://github.com/microsoft/react-native-windows/blob/master/docs/tracing.md ..

**Pull request merged**